dev/bots/analyze-sample-code.dart

```diff
@@ -314,18 +315,20 @@ linter:
   exit(exitCode);
 }
 
+final RegExp _constructorRegExp = new RegExp(r'[A-Z][a-zA-Z0-9<>.]*\(');
+
 int _expressionId = 0;
 
 void processBlock(Line line, List<String> block, List<Section> sections) {
   if (block.isEmpty)
     throw '$line: Empty ```dart block in sample code.';
-  if (block.first.startsWith('new ') || block.first.startsWith('const ')) {
+  if (block.first.startsWith('new ') || block.first.startsWith('const ') || block.first.startsWith(_constructorRegExp)) {
```

Comment:
do we still need "new" here?

Reply:
Yes otherwise the analyze fails because `new Class();` can not be at the top level of the library:

```dart
library xxx;

new Class(); // not valid dart code
```

Reply:
What I mean is that I'm surprised we still have any sample code that uses `new`.

Reply:
Without the `new` test the analysis output is really bad when a new is still there:

```/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:119:7: Expected an identifier (missing_identifier)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:119:7: Expected to find ';' (expected_token)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:119:7: Variables must be declared using the keywords 'const', 'final', 'var'
or a type name (missing_const_final_var_or_type)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:120:14: Named parameters must be enclosed in curly braces ('{' and '}') (named_parameter_outside_group)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:120:16: Default values of an optional parameter must be constant (non_constant_default_value)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:121:12: Named parameters must be enclosed in curly braces ('{' and '}') (named_parameter_outside_group)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:122:12: Named parameters must be enclosed in curly braces ('{' and '}') (named_parameter_outside_group)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:123:18: Named parameters must be enclosed in curly braces ('{' and '}') (named_parameter_outside_group)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:124:18: Named parameters must be enclosed in curly braces ('{' and '}') (named_parameter_outside_group)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:124:20: Default values of an optional parameter must be constant (non_constant_default_value)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:129:22: Named parameters must be enclosed in curly braces ('{' and '}') (named_parameter_outside_group)
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:129:24: Default values of an optional parameter must be constant (non_constant_default_value)
/tmp/flutter_analyze_sample_code.UHBKXK/main.dart:103:1: A function body must be provided
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:160:9: The named parameter 'onChangeEnd' isn't defined (undefined_named_parameter)
```

vs. (with the current code)

```
/home/me/dart/flutter/flutter/packages/flutter/lib/src/cupertino/slider.dart:119:7: Unnecessary new keyword (unnecessary_new)
```

Reply:
>Bonus points if you can find a way to test this, so that someone doesn't "optimise" the code later and remove the "new", thus making the experience worse.

I think it's already covered by [dev/bots/test/analyze-sample-code_test.dart](https://github.com/flutter/flutter/blob/5f4019367b2c0e906e9ab2acdcee12581241215d/dev/bots/test/analyze-sample-code_test.dart#L33)

Reply:
Fantastic.